### ISSUE TITLE
Problems when training machine translation with spm

### ISSUE THREAD

**Issue by vergilus** [google/sentencepiece#265]
Thanks for your work! I have been trying on en-fr machine translation task with sentencepiece.
According to the paper, sentencepiece should be no worse than BPE, however I found my training loss extremely unstable. 

![image](https://user-images.githubusercontent.com/19833549/50564067-0647d000-0d5d-11e9-8603-ee11726a5b68.png)

![image](https://user-images.githubusercontent.com/19833549/50564059-fb8d3b00-0d5c-11e9-8ebc-0ab7adb3b437.png)

the orange line is Rico Sennrich's BPE, the bleu one is sentencepiece.  This is the script I used for generating sentencepiece.

`  spm_train --input=/home/public_data/nmtdata/wmt15_en-fr/small_train/train_enfr.fr \
             --input_sentence_size=10000000 \
             --model_prefix=fr.bpe30k \
             --vocab_size=32768 \
             --input_format=text \
             --pad_id=0 \
             --eos_id=1 \
             --bos_id=2 \
             --unk_id=3 \
             --model_type=bpe`

I have been stuck on this problem for a week, desperate for some answers.

**Comment by taku910** [comment]
Thank you for using sentencepiece.

It would be hard to figure out the root cause only with this information.  For example, which nmt system are you using? How did you use sentencepiece?

**Comment by monsieurzhang** [comment]
It seems there is a data distribution problem to train a NMT model, if you just concatenate the two parts and then train with this data.
Can you try to shuf by yourself before feed them to the NMT system?

**Comment by vergilus** [comment]
@monsieurzhang 
I did shuffle the dataset and follow the tensor2tensor routine to generate a batch.

**Comment by taku910** [comment]
We are afraid that there is nothing we can do to help since this is an environment-dependent issue. if there is no further update, let me close this issue by the end of Aug.
